### PR TITLE
pkg/newlog: suppress log messages with newly introduced syslog/kernel log levels

### PR DESCRIPTION
**Pull request by rouming**
This PR pulls latest pillar definitions and implements syslog and kernel log suppression according to the 'debug.syslog.loglevel' and 'debug.kernel.loglevel' configuration properties.

### CODE REVIEW

**Review by naiming-zededa — APPROVED** [review]
LGTM

File: pkg/newlog/cmd/newlogd.go [hunk 1]
```diff
@@ -499,10 +514,27 @@ func handleGlobalConfigImp(ctxArg interface{}, key string, statusArg interface{}
...
+		atomic.StoreUint32(&syslogPrio, parseLogLevel(syslogPrioStr))
+
+		// parse kernel log level
+		kernelPrioStr := gcp.GlobalValueString(types.KernelLogLevel)
+		atomic.StoreUint32(&kernelPrio, parseLogLevel(kernelPrioStr))
 	}
 	log.Tracef("handleGlobalConfigModify done for %s, fastupload enabled %v", key, enableFastUpload)
 }
 
+func suppressMsg(entry inputEntry, cfgPrio uint32) bool {
```
Comment (eriknordmark) [inline pkg/newlog/cmd/newlogd.go:529]: should this be named supressKernelMsg? It is using SyslogKernelDefaultLogLevel

**Comment by rouming** [comment]
Rebased on latest master. Once #3807 is merged, I will rework correct sha in this one.

**Review by eriknordmark — COMMENTED** [review]
Why do you need cnirpc for this PR?

**Comment by milan-zededa** [comment]
> @milan-zededa Milan, could you please review this. I also added the following patch: [1b2f4fa](https://github.com/lf-edge/eve/commit/1b2f4fac127d04e5f9f5a015eb4f2de4b17dcc58) which fixes vendoring for the newlog, this repeats the kube/cnirpc line added in pillar/go.mod. I'm not sure that this is the right way, so I just copied what you did.

I forgot to remove that replace. It was needed because the package referenced from pillar was not yet merged (it was part of the same PR). I'm going to open a PR removing that. Then, once it is merged and you rebase your PR, you should no longer have this problem.

**Comment by milan-zededa** [comment]
@rouming We will need to merge this first: https://github.com/lf-edge/eve/pull/3839

**Comment by rouming** [comment]
> @rouming We will need to merge this first: #3839

Ok, now clean and tidy! Thanks.

**Review by milan-zededa — APPROVED** [review]

**Review by eriknordmark — APPROVED** [review]
LHTM

File: pkg/newlog/cmd/newlogd.go [hunk 2]
```diff
@@ -103,8 +104,12 @@ var (
 
 	//domainUUID
 	domainUUID *base.LockedStringMap // App log, from domain-id to appDomain
-	// syslog/kmsg priority string definition
-	priorityStr = [8]string{"emerg", "alert", "crit", "err", "warning", "notice", "info", "debug"}
+	// Default log levels for some subsystems. Variables are updated and used
+	// from different goroutines, so in order to push the changes ouf of the
```
Comment (rene) [inline pkg/newlog/cmd/newlogd.go:108]: small typo here: "outf" -> "out"

**Comment by rouming** [comment]
> There are build errors of the form: #10 [build 5/6] RUN GO111MODULE=on CGO_ENABLED=0 go build -ldflags "-s -w -X=main.Version=v0.0.0-20240403104824-8a3f5c38d3cb " -mod=vendor -o /out/usr/bin/newlogd ./cmd #10 0.173 go: finding module for package github.com/lf-edge/eve/pkg/kube/cnirpc #10 0.192 vendor/github.com/lf-edge/eve/pkg/pillar/types/domainmgrtypes.go:18:2: cannot query module due to -mod=vendor Error: error building "lfedge/eve-newlog:1f1c5b988504157e319b317455b3a5734d2640b7": error building for arch amd64: failed to solve: process "/bin/sh -c GO111MODULE=on CGO_ENABLED=0 go build -ldflags "-s -w -X=main.Version=${GOPKGVERSION}" -mod=vendor -o /out/usr/bin/newlogd ./cmd" did not complete successfully: exit code: 1 2024/04/10 13:25:08 error during command execution: error building "lfedge/eve-newlog:1f1c5b988504157e319b317455b3a5734d2640b7": error building for arch amd64: failed to solve: process "/bin/sh -c GO111MODULE=on CGO_ENABLED=0 go build -ldflags "-s -w -X=main.Version=${GOPKGVERSION}" -mod=vendor -o /out/usr/bin/newlogd ./cmd" did not complete successfully: exit code: 1 make: *** [Makefile:968: eve-newlog] Error 1

That's strange. I mean as usual I do not understand error messages from our build system, but I remember after Milan's PR I removed the cnirpc dependency and compiled locally successfully. I repeat the same: update with latest sha, go mod tidy, go mod vendor. (again no errors locally). Let's see this time.

**Comment by rouming** [comment]
It seems same error: "Error: error building "lfedge/eve-newlog:1c0af0a223bf3352fb0233bb8b0a029437098a48": error building for arch amd64: failed to solve: process "/bin/sh -c GO111MODULE=on CGO_ENABLED=0 go build -ldflags \"-s -w -X=main.Version=${GOPKGVERSION}\" -mod=vendor -o /out/usr/bin/newlogd ./cmd" did not complete successfully: exit code: 1" 

Unfortunately I do not understand how to fix the error without the error. Need help.

**Review by eriknordmark — APPROVED** [review]
LGTM

**Pull request merged**